pkg/framework/machines.go

```diff
@@ -44,6 +45,24 @@ func GetMachine(c client.Client, name string) (*mapiv1beta1.Machine, error) {
 	return machine, nil
 }
 
+// MachinesPresent search for each provided machine in `machines` argument in the predefined `existingMachines` list
+// and returns true when all of them were found
+func MachinesPresent(existingMachines []*mapiv1beta1.Machine, machines ...*mapiv1beta1.Machine) bool {
+	if len(existingMachines) < len(machines) {
+		return false
+	}
+	existingMachinesMap := map[types.UID]bool{}
+	for _, existing := range existingMachines {
+		existingMachinesMap[existing.UID] = true
+	}
```

Comment:
Technically this is more efficient, but what you have is fine, just as an education point, change if you want. A bool takes up more memory than an empty struct
```suggestion
	existingMachinesMap := map[types.UID]struct{}{}
	for _, existing := range existingMachines {
		existingMachinesMap[existing.UID] = struct{}
	}
```